version: '3.3'
services:
  bot:
    build: .
    depends_on:
      - postgres
  postgres:
    image: postgres:14.1-alpine
    restart: on-failure
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
    ports:
      - '5432:5432'
    volumes:
      - db:/var/lib/postgresql/data
      - ./db/init.sql:/docker-entrypoint-initdb.d/init.sql
  pgadmin:
    image: dpage/pgadmin4
    ports:
      - '9999:80'
    environment:
      - PGADMIN_DEFAULT_EMAIL=test@test.com
      - PGADMIN_DEFAULT_PASSWORD=test
    depends_on:
      - postgres
  s3:
    image: localstack/localstack:0.13.0.4-arm64
    container_name: s3
    network_mode: bridge
    environment:
      - AWS_DEFAULT_REGION=us-east-1
      - EDGE_PORT=4566
      - SERVICES=s3
      - DEBUG=1
      - DATA_DIR=/tmp/localstack/data
    ports:
      - '4566:4566'
    volumes:
     - "${LOCALSTACK_VOLUME_DIR:-./volume}:/var/lib/localstack"
     - "/var/run/docker.sock:/var/run/docker.sock"

volumes:
  db:
    driver: local
